pkg/progress/progressreader.go

```diff
@@ -20,10 +20,11 @@ type Reader struct {
 }
 
 // NewProgressReader creates a new ProgressReader.
-func NewProgressReader(in io.ReadCloser, out Output, size int64, id, action string) *Reader {
+func NewProgressReader(in io.ReadCloser, out Output, start, size int64, id, action string) *Reader {
```

Comment:
The new argument, `start`, is `0` in almost every call, pointing to the fact that it is likely not needed.

Let's add a `SetOffset` to the `progress.Reader` type. Also, this should always be called `offset`.